File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.py
import sys
n = int(sys.stdin.readline())

dp = [0] * (n + 1)


for i in range(n + 1):
    if (i < 3): dp[i] = i
    else:
        dp[i] = (dp[i - 1] + dp[i - 2]) % 10007 

print(dp[n])

File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n;

rl.on("line", (input) => {
  n = parseInt(input)
  
  rl.close()
});

// output
rl.on("close", () => {
  let dp = new Array(1001).fill(0)
  dp[1] = 1
  dp[2] = 2
  dp[3] = 3

  for (let i = 4; i < 1001; i++) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 10007
  }
  console.log(dp[n])
});